### PR TITLE
Wider/Deeper Surface Refinement MLP: hidden 256/384, layers 3/4

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The surface refinement MLP (`--surface_refine`) is our key pressure correction mechanism and a confirmed win in the baseline. Current config: `--surface_refine_hidden 192 --surface_refine_layers 3`.

More capacity in this correction head could enable finer-grained pressure pattern learning, especially for the OOD NACA6416 fore-foil in tandem configurations. Unlike backbone changes (which we've seen disrupt optimized attention routing), the SRF head operates post-backbone on surface node features — adding capacity here has no risk of disrupting the learned routing.

The SRF operates on a small subset of nodes (surface nodes only), so VRAM cost is minimal.

## Instructions

Run **2 seeds (42, 73)** for **each of 2 configs**, using `--wandb_group frieren-wider-srf`:

**Config A — wider (256 hidden, same 3 layers):**
```bash
cd cfd_tandemfoil && python train.py --agent frieren \
  --wandb_name "frieren/wider-srf-h256-l3" --wandb_group frieren-wider-srf \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 256 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias \
  --seed 42
```
(repeat with `--seed 73`)

**Config B — deeper (384 hidden, 3 layers):**
```bash
cd cfd_tandemfoil && python train.py --agent frieren \
  --wandb_name "frieren/wider-srf-h384-l3" --wandb_group frieren-wider-srf \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 384 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias \
  --seed 42
```
(repeat with `--seed 73`)

Run all 4 seeds in parallel (you have 4 GPUs). Report p_in, p_oodc, p_tan, p_re for each config (2-seed averages).

**If Config A or B beats baseline on p_tan, also try:**
- Config C: `--surface_refine_hidden 256 --surface_refine_layers 4` (wider + deeper)

## Baseline

| Metric | Baseline (2-seed avg) |
|--------|----------------------|
| p_in | 13.05 |
| p_oodc | 7.70 |
| **p_tan** | **28.60** |
| p_re | 6.55 |

Baseline W&B runs: d7l91p0x (seed 42, p_tan=28.9), j9btfx09 (seed 73, p_tan=28.3)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent <name> --wandb_name "<name>/baseline-gsb-pcgrad" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT frieren:

## Results

### Summary

**Both wider SRF configs are worse than baseline on p_tan.** Config A (h=256): p_tan +4.7%. Config B (h=384): p_tan +4.0%. Neither config meets the threshold for a Config C (wider+deeper) follow-up. Increasing SRF capacity hurts tandem transfer while marginally helping p_re.

### Metrics

| Config | Seed | p_in | p_oodc | p_tan | p_re | W&B |
|--------|------|------|--------|-------|------|-----|
| h=256, L=3 | 42 | 12.7 | 8.0 | 30.2 | 6.4 | `me8pq3ec` |
| h=256, L=3 | 73 | 13.9 | 7.6 | 29.7 | 6.4 | `068695h7` |
| **h=256 avg** | — | **13.3** | **7.8** | **29.95** | **6.4** | |
| h=384, L=3 | 42 | 13.3 | 7.8 | 29.6 | 6.6 | `b2yqrp7k` |
| h=384, L=3 | 73 | 13.7 | 7.7 | 29.9 | 6.4 | `h3r5gz4p` |
| **h=384 avg** | — | **13.5** | **7.75** | **29.75** | **6.5** | |
| **Baseline (h=192)** | — | 13.05 | 7.70 | 28.60 | 6.55 | d7l91p0x, j9btfx09 |

| Config | Δ p_in | Δ p_oodc | Δ p_tan | Δ p_re |
|--------|--------|----------|---------|--------|
| h=256 | +1.9% | +1.3% | **+4.7%** | -2.3% |
| h=384 | +3.4% | +0.6% | **+4.0%** | -0.8% |

### VRAM & Speed
- h=256: 44.0-46.2 GB, 71s/epoch (≈baseline)
- h=384: 41.8-46.7 GB, 70-79s/epoch (≈baseline)
- Both configs completed 150-151 epochs in ~180 min

### Commands

```bash
# Config A: h=256
cd cfd_tandemfoil && python train.py --agent frieren \
  --wandb_name "frieren/wider-srf-h256-l3-s42" --wandb_group frieren-wider-srf \
  --surface_refine_hidden 256 --surface_refine_layers 3 --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias

# Config B: same but --surface_refine_hidden 384
```

### What happened

Increasing SRF hidden dim from 192 to 256 or 384 consistently hurts p_tan by ~4-5%. The pattern is clear:
- **p_re improves** slightly (6.4-6.5 vs 6.55 baseline) — the wider SRF can better fit OOD-Re surface patterns
- **p_tan degrades** significantly (+4-5%) — more SRF capacity likely overfits to the surface patterns in the training set, harming generalization to the OOD NACA6416 fore-foil in tandem configurations
- **p_oodc** is flat to slightly worse
- **p_in** is slightly worse (more capacity = more overfitting risk)

The existing h=192 appears near-optimal: enough capacity for useful surface correction without overfitting. The SRF head operates post-backbone on surface features, so more capacity lets it memorize training-distribution surface pressure patterns rather than learning generalizable corrections.

Interestingly, h=384 avg is slightly better than h=256 avg on p_tan (29.75 vs 29.95). This could suggest that at even larger capacity, the model might regularize differently — but both are far from baseline and the trend is not promising.

### Suggested follow-ups

1. **Reduce SRF to h=128 or h=96:** If 192→256 hurts, the optimum might be smaller. Try reducing capacity.
2. **Regularize the SRF:** Add dropout (0.1-0.2) to the SRF MLP to prevent overfitting while keeping higher capacity.
3. **Condition-aware SRF:** Instead of wider hidden layers, try conditioning the SRF on (Re, AoA, gap, stagger) so it can specialize its corrections per-regime without memorizing.
4. **Not pursuing Config C** as instructed — neither Config A nor B beat baseline.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2170 because ADVISOR: Closing — both configs regress p_tan. h=256: +4.7%, h=384: +4.0%.

More SRF capacity overfits to training-distribution surface pressure patterns, harming NACA6416 generalization. h=192 is near-optimal — enough capacity for useful corrections without memorization.

The p_re improvement (-2.3% for h=256) is noted but doesn't offset the p_tan regression. Your analysis is correct: the SRF at 192 hidden is at the sweet spot between correction power and overfitting.

Dead end: SRF hidden dim changes. The 192/3-layer config is confirmed optimal.

Great work — the clean comparative setup makes this very clear..
